Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,511 @@
+        if (bookmarkComponent)
+        {
+            return bookmarkComponent;
+        }
+
+        // If we didn't find a component then we add it and return it.
+        levelEntity->Deactivate();
+        levelEntity->CreateComponent<LocalViewBookmarkComponent>();
+        levelEntity->Activate();
+        bookmarkComponent = levelEntity->FindComponent<LocalViewBookmarkComponent>();
+        
+        AZ_Assert(bookmarkComponent, "Couldn't create LocalViewBookmarkComponent.");
+        return bookmarkComponent;
+    }
+
+    AZStd::string LocalViewBookmarkLoader::GenerateBookmarkFileName() const
+    {
+        auto* prefabEditorEntityOwnershipInterface = AZ::Interface<AzToolsFramework::PrefabEditorEntityOwnershipInterface>::Get();
+        AZ_Assert(prefabEditorEntityOwnershipInterface != nullptr, "PrefabEditorEntityOwnershipInterface is not found.");
+        AzToolsFramework::Prefab::TemplateId rootPrefabTemplateId = prefabEditorEntityOwnershipInterface->GetRootPrefabTemplateId();
+
+        if (rootPrefabTemplateId == AzToolsFramework::Prefab::InvalidTemplateId)
+        {
+            return AZStd::string();
+        }
+
+        auto* prefabSystemComponent = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
+        AZ_Assert(
+            prefabSystemComponent != nullptr,
+            "Prefab System Component Interface could not be found. "
+            "It is a requirement for the LocalViewBookmarkLoader class. "
+            "Check that it is being correctly initialized.");
+
+        auto prefabTemplate = prefabSystemComponent->FindTemplate(rootPrefabTemplateId);
+        AZStd::string prefabTemplateName = prefabTemplate->get().GetFilePath().Filename().Stem().Native();
+
+        // To generate the file name in which we will store the view Bookmarks we use the name of the prefab + the timestamp
+        // e.g. LevelName_1639763579377.setreg
+        AZStd::chrono::system_clock::time_point now = AZStd::chrono::system_clock::now();
+        AZStd::string sTime = AZStd::string::format("%llu", now.time_since_epoch().count());
+        return prefabTemplateName.data() + AZStd::string("_") + sTime + ".setreg";
+    }
+
+    bool LocalViewBookmarkLoader::LoadDefaultLocalViewBookmarks()
+    {
+        // Write to the settings registry
+        auto registry = AZ::SettingsRegistry::Get();
+        if (!registry)
+        {
+            AZ_Warning("LocalViewBookmarkLoader", false, "Couldn't load Settings Registry");
+            return false;
+        }
+
+        LocalViewBookmarkComponent* bookmarkComponent = RetrieveLocalViewBookmarkComponent();
+        if (!bookmarkComponent)
+        {
+            AZ_Warning("LocalViewBookmarkLoader", false, "Couldn't find a LocalViewBookmarkComponent");
+            return false;
+        }
+
+        // if the field is not empty then we have a file linked to the prefab.
+        if (bookmarkComponent && !bookmarkComponent->GetLocalBookmarksFileName().empty())
```

Comment:
No need to check `bookmarkComponent` again, it was checked in the previous `if`.

Reply:
Done!